### PR TITLE
Use DispatcherQueue instead of CoreDispatcher for Mso::DispatchQueue

### PR DESCRIPTION
**Pull request by vmoroz**
The CoreDispatcher is a UWP-only implementation of a queue running on top of a window message handler. The newer Windows versions have new DispatcherQueue that works for UWP and Win32 applications.

In this PR we change the Mso::DispatchQueue to use the new DispatcherQueue instead of CoreDispatcher.  It enables use of the UI Mso::DispatchQueue implementation on Win32 platforms.

We had to change use of the UI Mso::DispatchQueue in our code to accommodate the DispatcherQueue restriction that it cannot be globally discovered in the application. Instead, we must create it in the ReactApplication and pass through the ReactNativeHost.InstanceSettings().Properties() to other components and native modules.

After this change we do not see the Playground Win32 crash at the startup.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/4877)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/IReactDispatcher.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#include "ReactDispatcherHelper.g.h"
+#include <ReactPropertyBag.h>
+#include <dispatchQueue/dispatchQueue.h>
+#include <winrt/Microsoft.ReactNative.h>
+
+namespace winrt::Microsoft::ReactNative {
+
+struct ReactDispatcher : implements<ReactDispatcher, IReactDispatcher> {
+  ReactDispatcher() = default;
+  ReactDispatcher(Mso::DispatchQueue &&queue) noexcept;
```
Comment (NikoAri) [inline vnext/Microsoft.ReactNative/IReactDispatcher.h:14]: >DispatchQueue [](start = 23, length = 13)

(nit) why not IDispatchQueue?

File: vnext/Microsoft.ReactNative/IReactDispatcher.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  bool HasThreadAccess() noexcept;
+  void Post(ReactDispatcherCallback const &callback) noexcept;
+
+  static Mso::DispatchQueue GetUIDispatchQueue(IReactPropertyBag const &properties) noexcept;
+
+  static IReactDispatcher UIThreadDispatcher() noexcept;
+  static ReactPropertyId<IReactDispatcher> UIThreadDispatcherProperty() noexcept;
+  static IReactDispatcher GetUIThreadDispatcher(IReactPropertyBag const &properties) noexcept;
+  static void SetUIThreadDispatcher(IReactPropertyBag const &properties) noexcept;
```
Comment (NikoAri) [inline vnext/Microsoft.ReactNative/IReactDispatcher.h:24]: >SetUIThreadDispatcher [](start = 14, length = 21)

SetUIThreadDispatcherProperties?

File: vnext/Microsoft.ReactNative/IReactDispatcher.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+struct ReactDispatcher : implements<ReactDispatcher, IReactDispatcher> {
+  ReactDispatcher() = default;
+  ReactDispatcher(Mso::DispatchQueue &&queue) noexcept;
+
+  bool HasThreadAccess() noexcept;
+  void Post(ReactDispatcherCallback const &callback) noexcept;
+
+  static Mso::DispatchQueue GetUIDispatchQueue(IReactPropertyBag const &properties) noexcept;
+
+  static IReactDispatcher UIThreadDispatcher() noexcept;
```
Comment (NikoAri) [inline vnext/Microsoft.ReactNative/IReactDispatcher.h:21]: >UIThreadDispatcher [](start = 26, length = 18)

Should this match api on line 23? Eg: GetUIThreadDispatcher? or the other way around?

Comment (NikoAri) [inline vnext/Microsoft.ReactNative/IReactDispatcher.h:21]: >IReactDispatcher [](start = 9, length = 16)

shouldn't this be a pointer? Or winrt idl is smart for this type of object?

File: vnext/Microsoft.ReactNative/IReactDispatcher.h [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+#include <winrt/Microsoft.ReactNative.h>
+
+namespace winrt::Microsoft::ReactNative {
+
+struct ReactDispatcher : implements<ReactDispatcher, IReactDispatcher> {
+  ReactDispatcher() = default;
+  ReactDispatcher(Mso::DispatchQueue &&queue) noexcept;
+
+  bool HasThreadAccess() noexcept;
+  void Post(ReactDispatcherCallback const &callback) noexcept;
```
Comment (NikoAri) [inline vnext/Microsoft.ReactNative/IReactDispatcher.h:17]: >& [](start = 42, length = 1)

Why not &&?

File: vnext/Microsoft.ReactNative/IReactDispatcher.h (hunk 2 above)
Comment (NikoAri) [inline vnext/Microsoft.ReactNative/IReactDispatcher.h:24]: >SetUIThreadDispatcher [](start = 14, length = 21)

this is weird, why not just have SetUIThreadDispatcher(IReactDispatcher)?
that way users can do
SetUIThreadDispatcher(GetUIThreadDispatcher(properties))
or am I misunderstanding this Api?

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    // Post task for the asynchronous execution.
+    void Post(ReactDispatcherCallback callback);
+  }
+
+  // Helper methods for the property bag implementation.
+  [webhosthidden]
+  static runtimeclass ReactDispatcherHelper
+  {
+    // Get or create IReactDispatcher for the current UI thread.
+    static IReactDispatcher UIThreadDispatcher{ get; };
```
Comment (NickGerleman) [inline vnext/Microsoft.ReactNative/IReactDispatcher.idl:27]: Is this exposed to consumers? From what I understood we weren't sure yet how we wanted to treat the multi-UI-thread scenario. I worry that anything we do expose would be hard to walk back.

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+    // Get name of the UIThreadDispatcher property.
+    static IReactPropertyName UIThreadDispatcherProperty();
```
Comment (NickGerleman) [inline vnext/Microsoft.ReactNative/IReactDispatcher.idl:30]: Same here. Not sure we should expose what we consider to be the UI thread yet. People can already do this manually with the property bag, right?

File: vnext/Mso/dispatchQueue/dispatchQueue.h [hunk 7]
```diff
@@ -545,10 +538,6 @@ inline /*static*/ DispatchQueue const &DispatchQueue::ConcurrentQueue() noexcept
   return IDispatchQueueStatic::Instance()->ConcurrentQueue();
 }
 
-inline /*static*/ DispatchQueue const &DispatchQueue::MainUIQueue() noexcept {
```
Comment (NikoAri) [inline vnext/Mso/dispatchQueue/dispatchQueue.h:548]: >MainUIQueue [](start = 54, length = 11)

Isn't this a break from DevMain? Would DevMain use new pattern in the future?

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl (hunk 6 above)
Comment (NickGerleman) [reply vnext/Microsoft.ReactNative/IReactDispatcher.idl:30]: If we do want to directly expose a thread to the consumers and its available in all cases, maybe we should expose it on the context + InstanceSettings without letting people know it's in the bag? From the outside observer perspective, ReactHost in Office using the property bag for queues felt like it was hiding something meant to be explicit.

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj.filters [hunk 8]
```diff
@@ -648,7 +649,6 @@
     <ClInclude Include="ReactHost\UwpReactInstanceProxy.h">
       <Filter>ReactHost</Filter>
     </ClInclude>
-    <ClInclude Include="ReactPropertyBag.h" />
```
Comment (NickGerleman) [inline vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj.filters:651]: Why is this removed?

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+import "IReactPropertyBag.idl";
+
+namespace Microsoft.ReactNative {
+
+  // The delegate is used to create property value on-demand.
+  [webhosthidden]
+  delegate void ReactDispatcherCallback();
+
+  [webhosthidden]
+  interface IReactDispatcher
```
Comment (aeulitz) [inline vnext/Microsoft.ReactNative/IReactDispatcher.idl:13]: Can this replace the IMessageQueue interface (see Desktop/ABI/MessageQueue.idl)?

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl (hunk 5 above)
Comment (NickGerleman) [reply vnext/Microsoft.ReactNative/IReactDispatcher.idl:27]: Oh, I think I misinterpreted the API. If this is just for getting the dispatcher of current thread that makes sense.

File: vnext/Microsoft.ReactNative/IReactDispatcher.h (hunk 1 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.h:14]: The open source implementation of the DispatchQueue is different. We do not use the interface directly anymore. Instead, the DispatchQueue is a wrapper of the interface similar to what we do with the Mso::Future or Mso::Promise.
The good thing about having a wrapper class like Mso::DispatchQueue is that we can have template methods there, static methods, and all the DispatchQueue APIs are grouped under one class that makes it IntelliSense friendly.

---
In reply to: [423963640](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423963640) [](ancestors = 423963640)

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl (hunk 5 above)
Comment (aeulitz) [inline vnext/Microsoft.ReactNative/IReactDispatcher.idl:27]: >get [](start = 48, length = 3)

For my understanding, who sets this? Does the API design represented here accommodate user who want to provide there own dispatcher implementations?

File: vnext/Microsoft.ReactNative/IReactDispatcher.h (hunk 4 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.h:17]: This is C++/WinRT thing : they pass parameters by const&. I will check again if we can use && in places like this, but my previous tries were not successful.

---
In reply to: [423967144](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423967144) [](ancestors = 423967144)

File: vnext/Microsoft.ReactNative/IReactDispatcher.h (hunk 1 above)
Comment (aeulitz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.h:14]: Does the mechanism you describe above allow people to provide their own dispatcher implementations?

---
In reply to: [423982263](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423982263) [](ancestors = 423982263,423963640)

File: vnext/Microsoft.ReactNative/IReactDispatcher.h (hunk 3 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.h:21]: C++/WinRT interface-looking types are actually smart pointers to a real ABI interface.

---
In reply to: [423966942](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423966942) [](ancestors = 423966942)

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl (hunk 5 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.idl:27]: This property is per thread. It does not matter what we decide in future: this code will work in any case. There is no such thing as 'perfect API'. This design works today. Tomorrow if we have new requirements we will change it.

---
In reply to: [423970645](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423970645) [](ancestors = 423970645)

**Comment by vmoroz** [comment]
The main goal at this point is to supply UI dispatch queue from outside of ReactNativeHost. The ReactNativeHost cannot create it on its own because it may be created in a background thread. Thus, the UI thread must be passed explicitly.

---
In reply to: [410353268](https://github.com/microsoft/react-native-windows/pull/4877#issuecomment-410353268) [](ancestors = 410353268)

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    // True if dispatcher uses current thread.
+    Boolean HasThreadAccess { get; };
+
+    // Post task for the asynchronous execution.
+    void Post(ReactDispatcherCallback callback);
+  }
+
+  // Helper methods for the property bag implementation.
+  [webhosthidden]
+  static runtimeclass ReactDispatcherHelper
```
Comment (aeulitz) [inline vnext/Microsoft.ReactNative/IReactDispatcher.idl:24]: >static runtimeclass ReactDispatcherHelper [](start = 2, length = 41)

Nit: Should we keep it at one type per file?

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl (hunk 5 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.idl:27]: Yes, this is about getting dispatcher of the current thread. The UIThreadDispatcherProperty can be used to set the UIThreadDispatcher in the ReactPropertyBag.

---
In reply to: [423981509](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423981509) [](ancestors = 423981509)

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl (hunk 10 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.idl:24]: No, we do not have such requirement. Seeing how slow the MIDL compiler is, I would say the opposite: less IDL files we have is better. I would rather use the IDL files as a group of definitions related to some functionality.

---
In reply to: [423987808](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423987808) [](ancestors = 423987808)

File: vnext/ReactUWP/Modules/AppearanceModule.h [hunk 11]
```diff
@@ -17,13 +17,14 @@ class AppearanceChangeListener final : public Mso::ActiveObject<> {
...
  public:
-  AppearanceChangeListener(std::weak_ptr<IReactInstance> &&reactInstance) noexcept;
+  AppearanceChangeListener(std::weak_ptr<IReactInstance> &&reactInstance, Mso::DispatchQueue const &uiQueue) noexcept;
   const char *GetColorScheme() const noexcept;
 
  private:
   static const char *ToString(ApplicationTheme theme) noexcept;
   void OnColorValuesChanged() noexcept;
 
+ private:
```
Comment (aeulitz) [inline vnext/ReactUWP/Modules/AppearanceModule.h:27]: >private [](start = 1, length = 7)

Nit: Why the repetition of the 'private:' marker?

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl (hunk 5 above)
Comment (NickGerleman) [reply vnext/Microsoft.ReactNative/IReactDispatcher.idl:27]: @vmoroz are external native modules expected to use this? "Tomorrow if we have new requirements we will change it." becomes painful if users start relying on this.

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl (hunk 6 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.idl:30]: Good point. We may do it in future. For now I would prefer it to be in the property bag. There are several reasons:
- We want to associate a queue with a native module. The best way to do it is to associate property ID with the native module. This property ID can be used to retrieve the real queue before we make instance of the module. It is the main reason why we are doing it in Office for Reka services.
- I do not want the ReactContext become too big or too fat. E.g. today it depends on XAML for view events and it already creates a lot of issues. Ideally, the context must be as small as possible. Its main goal is to be a central point to access shared data and resources from native modules. Ideally no preference must be given to any specific interface or component.

---
In reply to: [423973960](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423973960) [](ancestors = 423973960)

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl (hunk 5 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.idl:27]: Yes, the current vision is that developers of native modules can access IReactDispatcher of current UI thread using this API. This is a new API. I cannot give you 100% guarantee that this is the final one. It must tested by real usage first. I do not believe we any new API where we can provide such guarantee. I am not sure what is your concern here. Did you ever create API that you never changed?

---
In reply to: [423993539](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423993539) [](ancestors = 423993539)

File: vnext/Microsoft.ReactNative/IReactDispatcher.h (hunk 1 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.h:14]: In the new Mso::DispatchQueue model we have only one Mso::DispatchQueue class that manages queue of tasks. Then we have IDispatchQueueScheduler interface that can run these tasks according to specific rules on the threads it wants. It is relatively small interface and developers can provide their own implementation of it. By implementing custom IDispatchQueueScheduler interface, developers can implement a 'custom queue' when they give it to the Mso::DispatchQueue constructor.

---
In reply to: [423983579](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423983579) [](ancestors = 423983579,423982263,423963640)

File: vnext/Microsoft.ReactNative/IReactDispatcher.h (hunk 3 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.h:21]: These are tow different APIs:
- UIThreadDispatcher returns IReactDispatcher associated with the current UI thread.
- GetUIThreadDispatcher is an 'attached property' method that returns UIThreadDispatcher stored in the property bag.

---
In reply to: [423966470](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423966470) [](ancestors = 423966470)

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl (hunk 6 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.idl:30]: There must be a way to put something in the property bag.  In simple UWP 'green' applications we already do it in the ReactApplication. But for win32 XAML islands, we cannot use it and we must provide ability:
- What to put as a UI dispatcher in the property bag - UIThreadDispatcher property.
- How to put it in the property bag - use the UIThreadDispatcherProperty to store the UIThreadDispatcher.

---
In reply to: [423971383](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423971383) [](ancestors = 423971383)

File: vnext/Mso/dispatchQueue/dispatchQueue.h (hunk 7 above)
Comment (vmoroz) [reply vnext/Mso/dispatchQueue/dispatchQueue.h:548]: This code is not in the DevMain. It is part of the experimental DispatchQueue 2.0 implementation. At some point I thought that it is a great idea to have such global property, but not any more because we have scenarios in RNW where it does not work. 
So, this experimental feature will never land to DevMain.

---
In reply to: [423973832](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423973832) [](ancestors = 423973832)

File: vnext/Microsoft.ReactNative/IReactDispatcher.cpp [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "IReactDispatcher.h"
+#include "ReactDispatcherHelper.g.cpp"
+
+using namespace winrt;
+using namespace Windows::Foundation;
+
+namespace winrt::Microsoft::ReactNative {
```
Comment (JunielKatarn) [inline vnext/Microsoft.ReactNative/IReactDispatcher.cpp:11]: Why prefix these namespaces `winrt::`?

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj.filters (hunk 8 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj.filters:651]: It was removed automatically by VS. We do not have such files. The implementation files are IReactPropertyBag.h and IReactPropertyBag.cpp. I never can understand what rules VS uses to change the .filters files.


---
In reply to: [423977037](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423977037) [](ancestors = 423977037)

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl (hunk 9 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.idl:13]: I guess, yes. I was not aware about it.

---
In reply to: [423980262](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423980262) [](ancestors = 423980262)

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl (hunk 5 above)
Comment (NickGerleman) [reply vnext/Microsoft.ReactNative/IReactDispatcher.idl:27]: Of course we will sometimes need to make changes. It's just much more expensive to do this for the public API of a package where we don't control consumers.

For internal code, or if we're in a monorepo setup like Office, we can just fixup old code using an API we don't like. It's much more expensive to do this for RNW. We either make sudden breaks, which has been a big cause of customer frustration, or we need to do phased deprecation, which is a but less painful, but annoying to track and maintain.

In this instance, we know we have the potential to run into some issues with the current design. It feels like we could avoid pain if we can prevent these rapid changes. E.g. expose this internally for now.

Consumers can still manually inject a queue of their own property if they're okay with it being instance global. Directing people towards that would avoid needing to deal with breaks and user frustration if we decide we can do something better.

Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.idl:27]: This is just a shortcut to get a default implementation out of the current UI thread.
The current design has a gap that I do not understand yet how we can supply custom Office implementation. We must go back to solve it.
E.g. one of the solutions could be to make this property settable to enable such scenarios.

---
In reply to: [423982313](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423982313) [](ancestors = 423982313)

Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.idl:27]: "In this instance, we know we have the potential to run into some issues with the current design." Could you please share what you see as an issue?
Previous design was not multi UI thread friendly. This one makes it more friendly, but we still must provide a dedicated UI thread per RNH. It is not a 'true' multi-UI concept, but we must do it today because many native components and RN itself needs it. So, as any other engineering solution we must implement some kind of 'transitional' design. But even if multi-UI will never happen this current design is still much better and more usable comparing to what we have today.

I would like to understand what type of design issue you see that makes you block this PR.

---
In reply to: [424009257](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r424009257) [](ancestors = 424009257)

Comment (NickGerleman) [reply vnext/Microsoft.ReactNative/IReactDispatcher.idl:27]: Wasn't sure how long-term multi-ui thread was. From the last discussion I was under the impression there was some active effort there, but now it sounds like it's further out. Just wanting to make sure we don't break this too quickly or box ourselves into a corner.

Signing off.

File: vnext/Microsoft.ReactNative/IReactDispatcher.cpp (hunk 12 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.cpp:11]: The `winrt::` prefix is commonly used for the C++/WinRT components. It is used by default by the C++/WinRT code generator. It is much easier to author our own components in that namespace. Plus, we can use many winrt functions without prefixes.
Do you think it would be preferable not to use it?

---
In reply to: [424007290](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r424007290) [](ancestors = 424007290)

File: vnext/ReactUWP/Modules/AppearanceModule.h (hunk 11 above)
Comment (NickGerleman) [reply vnext/ReactUWP/Modules/AppearanceModule.h:27]: Could we leave this as is? I prefer the way I originally wrote this, and don't see why we should add our preferences to other code.

File: vnext/Microsoft.ReactNative/IReactDispatcher.idl (hunk 5 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.idl:27]: Thank you for the sign off!
As we discussed offline before, the core issue with the multi-UI threads are the core native modules and the core RN that relies on the single UI thread. 
This change allows us to bring the UI queue to the RNH as an external entity. The RNH does not create it anymore. This design change moves us closer to the multi-UI threads if we ever do it. Plus, it enables use of the RNH for the XAML islands.

---
In reply to: [424022761](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r424022761) [](ancestors = 424022761)

File: vnext/Microsoft.ReactNative/IReactDispatcher.h (hunk 4 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IReactDispatcher.h:17]: I tried it. It does not compile. The C++/WinRT uses const&.

---
In reply to: [423983290](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423983290) [](ancestors = 423983290,423967144)

File: vnext/ReactUWP/Modules/AppearanceModule.h (hunk 11 above)
Comment (vmoroz) [reply vnext/ReactUWP/Modules/AppearanceModule.h]: Removed it.

---
In reply to: [424025014](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r424025014) [](ancestors = 424025014)

**Comment by christophpurrer** [comment]
@vmoroz > Will this also be backported to 0.61?
